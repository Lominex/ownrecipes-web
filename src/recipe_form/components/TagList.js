import { Component } from 'react';

import { Input } from '../../common/components/FormComponents';

class TagList extends Component {
  unarrayify = value => value.map(tag => tag.title).join(', ');

  arrayify = value => {
    const dict = [];
    if (value) {
      const tags = value.split(',');
      // eslint-disable-next-line no-restricted-syntax, guard-for-in
      for (const title in tags) {
        dict.push({ title: tags[title].trim() });
      }
    }
    return dict;
  };

  handleChange = (name, value) => {
    if (this.props.change) {
      this.props.change(name, this.arrayify(value));
    }
  };

  render() {
    return (
      <Input
          name={this.props.name}
          label={this.props.label}
          placeholder={this.props.placeholder}
          size={this.props.size}
          change={this.handleChange}
          value={this.props.tags ? this.unarrayify(this.props.tags) : ''}
          errors={this.props.errors}
      />
    );
  }
}

export default TagList;
