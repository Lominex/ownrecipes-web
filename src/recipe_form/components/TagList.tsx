import Input from '../../common/components/Input';
import { Tag } from '../../recipe/store/RecipeTypes';

export interface ITagListProps {
  name:   string;
  label:  string;
  tooltip?: string;

  tags: Array<Tag>;
  errors?:  string;

  className?: string;
  onChange: (name: string, value: unknown) => void;
}

function stringify(value: Array<Tag>): string {
  return value.map(tag => tag.title).join(', ');
}

function arrayify(value: string): Array<Tag> {
  const dict: Array<Tag> = [];

  const tagsArray = value.split(',');
  tagsArray.forEach(title => {
    // TODO This seems hackish, as the id will be lost.
    dict.push({ title: title.trim() } as Tag);
  });

  return dict;
}

const TagList: React.FC<ITagListProps> = ({
    name, label, tooltip,
    tags, errors, className, onChange }: ITagListProps) => {
  const handleChange = (inpName: string, value: string) => {
    onChange(inpName, arrayify(value));
  };

  return (
    <Input
        name      = {name}
        label     = {label}
        tooltip   = {tooltip}
        value     = {stringify(tags)}
        errors    = {errors}
        className = {className}
        change    = {handleChange}
    />
  );
};

export default TagList;
