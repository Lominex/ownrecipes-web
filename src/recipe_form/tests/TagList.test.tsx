import renderer from 'react-test-renderer';

import TagList from '../components/TagList';

test('TagList component test', () => {
  const component = renderer.create(
    <TagList name='tag-list' label='TagList' tags={[]} onChange={() => { /* nothing */ }} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
