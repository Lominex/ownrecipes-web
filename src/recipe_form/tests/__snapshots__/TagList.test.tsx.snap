// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList component test 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="tag-list"
  >
    TagList
  </label>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      id="tag-list"
      name="tag-list"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;
